tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Test the Logitech Harmony Hub config flow."""
+import asyncio
+import json
+
+from asynctest import CoroutineMock, MagicMock, patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.hunterdouglas_powerview.const import DOMAIN
+
+from tests.common import load_fixture
+
+
+def _get_mock_powerview_userdata(userdata=None, get_resources=None):
+    mock_powerview_userdata = MagicMock()
+    if not userdata:
+        userdata = json.loads(load_fixture("hunterdouglas_powerview/userdata.json"))
+    if get_resources:
+        type(mock_powerview_userdata).get_resources = CoroutineMock(
+            side_effect=get_resources
+        )
+    else:
+        type(mock_powerview_userdata).get_resources = CoroutineMock(
+            return_value=userdata
+        )
+    return mock_powerview_userdata
+
+
+async def test_user_form(hass):
+    """Test we get the user form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    mock_powerview_userdata = _get_mock_powerview_userdata()
+    with patch(
+        "homeassistant.components.hunterdouglas_powerview.UserData",
+        return_value=mock_powerview_userdata,
+    ), patch(
+        "homeassistant.components.hunterdouglas_powerview.async_setup",
+        return_value=True,
+    ) as mock_setup, patch(
+        "homeassistant.components.hunterdouglas_powerview.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.2.3.4"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "AlexanderHD"
+    assert result2["data"] == {
+        "host": "1.2.3.4",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result3 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result3["type"] == "form"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5444210)